src/scripts/OSFramework/OSUI/Pattern/Video/Video.ts

```diff
@@ -0,0 +1,344 @@
+
+			this._setLoop();
+
+			this._setMuted();
+
+			this._setPosterUrl();
+
+			this._setWidth();
+
+			this._setHeight();
+
+			// Trigger the event of unstarted based on video progress
+			if (this._videoElement.currentTime === 0) {
+				this._triggerOnStateChangedEvent(Patterns.Video.Enum.VideoStates.Unstarted);
+			}
+		}
+
+		// Method create the source element
+		private _setVideoSource(): void {
+			// Get the file extension from URL
+			const urlFileExtension = this.configs.URL.split('.').pop();
+
+			// Add class to video source element
+			OSUI.Helper.Dom.Styles.AddClass(this._videoSourceElement, Patterns.Video.Enum.CssClass.VideoSource);
+
+			// Set the attributes to video source created
+			this._videoSourceElement.src = this.configs.URL;
+			this._videoSourceElement.type = Patterns.Video.Enum.VideoAttributes.TypePath + urlFileExtension;
+		}
+
+		// Method create the track element
+		private _setVideoTrack(): void {
+			// Check if contains tracks to be added
+			// If true, create the element with all the attributes
+			const subtitlesList = JSON.parse(this.configs.Subtitles);
+
+			if (subtitlesList.length > 0) {
+				for (const item of subtitlesList) {
+					const trackElement = document.createElement(Patterns.Video.Enum.VideoTags.Track);
+
+					// Add class to track element created
+					OSUI.Helper.Dom.Styles.AddClass(trackElement, Patterns.Video.Enum.CssClass.VideoTrack);
+
+					// Set track element attributes
+					trackElement.kind = Patterns.Video.Enum.VideoAttributes.Captions;
+					trackElement.srclang = item.LanguageCode;
+					trackElement.src = item.SourceFile;
+					trackElement.label = item.Label;
+
+					// Append the track element into video HTML element
+					this.selfElement.appendChild(trackElement);
+				}
+			}
+		}
+
+		// Method that will set the Width attribute
+		private _setWidth(): void {
+			if (this.configs.Width !== Constants.EmptyString) {
+				OSUI.Helper.Dom.Attribute.Set(
+					this._videoElement,
+					Patterns.Video.Enum.VideoAttributes.Width,
+					this.configs.Width
+				);
+			} else {
+				OSUI.Helper.Dom.Attribute.Remove(this._videoElement, Patterns.Video.Enum.VideoAttributes.Width);
+			}
+		}
+
+		// Method that triggers the OnStateChanged event
+		private _triggerOnStateChangedEvent(stateChanged: string): void {
+			this.triggerPlatformEventCallback(this._platformEventOnStateChanged.bind(this), stateChanged);
+		}
+
+		/**
+		 * Sets the A11Y properties when the pattern is built.
+		 *
+		 * @protected
+		 * @memberof OSFramework.Patterns.Video.Video
+		 */
+		protected setA11YProperties(): void {
+			console.log(GlobalEnum.WarningMessages.MethodNotImplemented);
+		}
+
+		/**
+		 * Sets the callbacks to be used in the pattern.
+		 *
+		 * @protected
+		 * @memberof OSFramework.Patterns.Video.Video
+		 */
+		protected setCallbacks(): void {
+			// Check the video time and trigger the event if is 0 with the status Unstarted
+			this._videoElement.onplay = () => {
```

Comment:
Why this one has a inline function call, instead of just passing a callback? Can't we have the If validation on the callback itself?

Reply:
I created a validation inside the callback that will run only based on the Unstarted state.